locale/en/docs/guides/dont-block-the-event-loop.md

```diff
@@ -0,0 +1,135 @@
+I lied.
+There is one aspect of V8 that is actually pretty slow.
+[*Regular expressions*](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions), while really handy, can actually take exponential-time.
+For example, if you are serving files and ensuring a path is valid, you might write "`if (f.match(/(\/.+)+$/) ...`" to make sure the path is a sequence of "/"-delimited names.
+Sadly, what if a client sends in the path `///.../\n` (say, 100 /'s followed by a newline character that the regexp's "." won't match)?
+Your server will take an extremely long time to evaluate the regexp with this input, and if the attacker adds another "/" the time taken will double.
+Try it!
+
+So why is this happening?
+The problem is that your server is using a "vulnerable regular expression", one that exposes you to Regular Expression Denial of Service ([ReDoS](https://www.owasp.org/index.php/Regular_expression_Denial_of_Service_-_ReDoS)).
+For this reason, you should be leery of using regular expressions to validate user input.
+
+There are some tools to check your regexps for safety, like
+- [safe-regex](https://github.com/substack/safe-regex)
+- [rxxr2](http://www.cs.bham.ac.uk/~hxt/research/rxxr2/)
+However, neither of these will catch all vulnerable regexps.
+
+A safer alternative is the [node-re2](https://github.com/uhop/node-re2) project, which uses Google's blazing-fast [RE2](https://github.com/google/re2) regexp engine. But be warned, RE2 is not 100% compatible with Node's regexps, so check for regressions if you swap in the node-re2 module to handle your regexps. And really complicated regexps may not be supported by node-re2.
+
+As usual, if you're trying to match something "obvious", like a URL or a file path, use an npm module!
+
+### I'm not using regexps but I still wanted to do something complicated in JavaScript
+OK, so your callback gets some input and you want to do something fancy with it. Maybe you're applying a function to every item in an array, and you're not sure how much that function costs.
+
+Here are some questions you should ask yourself:
+- Is Node really a good fit for my project? Node excels for I/O-bound work, but for expensive computation it might not be the best option.
+- Do I feel confident enough in my C++ skills to write a [C++ addon](https://nodejs.org/api/addons.html)? If so, consider offloading this work to the Worker Pool. But see the next section for more on that.
+- Would it be better to ask a [Child Process](https://nodejs.org/api/child_process.html) to do this for me?
+
+If you're in a hurry, though, here's a two-step way to handle this situation:
+1. Do part of your task, and then...
+2. defer the rest until later, e.g. with `setImmediate`.
+
+This gives all of the other pending clients a turn.
+It's pretty easy to accomplish this by sticking whatever state you're carrying around in a closure or an object.
+And you might want to add a TODO to revisit this section of your code later.
```

Comment:
Some example might be good here. I think most newcomers/basic users won't necessarily have a clue about what you're talking here. The notion of 'immediate' in the event loop can be misleading and some people won't understand why it would partly solve the problem IMHO.

Reply:
Good idea. I've been thinking about some examples.